pkg/operator/resource/resourceapply/generic.go

```diff
@@ -85,3 +155,23 @@ func ApplyDirectly(kubeClient kubernetes.Interface, recorder events.Recorder, ma
 
 	return ret
 }
+
+func (c *ClientHolder) configMapsGetter() corev1client.ConfigMapsGetter {
+	if c.kubeClient == nil {
+		return nil
+	}
+	if c.kubeInformers == nil {
+		return c.kubeClient.CoreV1()
+	}
+	return v1helpers.CachedConfigMapGetter(c.kubeClient.CoreV1(), c.kubeInformers)
+}
+
+func (c *ClientHolder) secretsGetter() corev1client.SecretsGetter {
+	if c.kubeClient == nil {
+		return nil
+	}
+	if c.kubeInformers == nil {
+		return c.kubeClient.CoreV1()
+	}
+	return v1helpers.CachedSecretGetter(c.kubeClient.CoreV1(), c.kubeInformers)
```

Comment:
I am still against this `corev1client.SecretsGetter` interface for the purpose of caching. The wrapper throws away the `GetOptions`. Instead make a smaller interface implemented by a more powerful implementation (clientset).

Reply:
non-overlapping sets makes adoption and usage harder,  but I'm happy to let this linger and take the rest.